### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by a-laarabi**
# What?
> In this repository, I configured My Redux Store and write actions and reducer for adding and removing books.

### Highlights:

- There are [no linter errors](https://github.com/microverseinc/linters-config).
- I used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- I documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- I did understand the concept of state management.
- I used store, actions, and reducers in React.

Cheer and happy reviewing 🥇

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    case ADD_BOOK:
+      return [
+        ...state,
+        action.book,
+      ];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.index);
+    default:
+      return state;
+  }
```
Comment (Ugboaja-Uchechi) [inline src/redux/books/books.js:17]: - [ ] Great job with your code 👍 . Unfortunately the functionality for adding and removing books doesn't work. When I try adding a book it updates in the search bar instead. Kindly fix this 🙏 

image

**Review by Ugboaja-Uchechi — CHANGES_REQUESTED** [review]
Hi @a-laarabi,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by a-laarabi** [comment]
heyy @Ugboaja-Uchechi 
thanks for your time it's a perfect review I want to mention that you probably forgot about this part from the [description of the project](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md) :

In this step, you will configure your Redux Store and write actions and reducer for adding and removing books. NOTE: editing an existing book and updating the progress is not part of this step. Neither is the styling of the application's components but adding and removing actions need to be implemented.

**Review by Lameck1 — COMMENTED** [review]
Hello @a-laarabi 👋,

_This is an additional review as a result of the second opinion request._

I would to confirm that your concern is valid. In this milestone, you are not supposed to implement any change in the components.

@Ugboaja-Uchechi, please take note of this, that `no changes in React components are required in this step`. It, therefore, means that the functionalities you have tested cannot work at the very moment until they are worked on in a milestone ahead.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please, request another code review and the next code reviewer will be able to pick it up from there.

Keep rocking 🚀!

**Review by descholar-ceo — APPROVED** [review]
Hi @a-laarabi 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 You are ready to proceed with the next milestone ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**